src/entity/Datafile.ts

```diff
@@ -1,16 +1,32 @@
-import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, BaseEntity } from 'typeorm';
+import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, BaseEntity, ManyToOne, JoinColumn } from 'typeorm';
+
+// eslint-disable-next-line import/no-cycle
+import { Dataset } from './dataset';
 
 @Entity({ name: 'datafiles' })
 export class Datafile extends BaseEntity {
     @PrimaryGeneratedColumn('uuid')
     id: string;
 
     @Column({ nullable: false })
-    name: string;
+    sha256hash: string;
 
-    @Column({ nullable: false })
-    description: string;
+    @ManyToOne(() => Dataset, (dataset) => dataset.datafiles, { onDelete: 'CASCADE', eager: true })
```

Comment:
So we don't yet have code to delete a dataset or a datafile for that matter... I believe if you're going to delete a dataset then you need to delete everything.  Equally I haven't settled on do you delete something or soft delete it where you just mark it as no longer active.

We've yet to make these implementation decisions.  I believe that if we had a delete method it would work like this:

```typescript
const datafiles = await dataset.datafiles;
for(const dfile in datafiles) {
    dataLakeService.delete(`${dfile.id}.csv`);
    dfile.detele();
}
const lookuptables = await dataset.lookuptables;
for(const ltable in datafiles) {
    dataLakeService.delete(`${ltable.id}.csv`);
    ltable.detele();
}
dataset.delete(); // Remove everything else which isn't stored in the datalake
```

But we're not there yet... I think this will come at some point soon though just for house keeping purposes after auth and rbac.